haystack/nodes/reader/table.py

```diff
@@ -32,6 +32,70 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseTableProcessor:
+    def preprocess(self, docs: List[Document]):
+        return docs
+
+    def postprocess(self, answers: List[Answer], docs: List[Document]):
+        return answers
+
+
+class TableProcessor(BaseTableProcessor):
+    def __init__(self, num_rows=7, num_cols=7, num_row_fixed=0, num_col_fixed=0) -> None:
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+        self.num_row_fixed = num_row_fixed
+        self.num_col_fixed = num_col_fixed
+
+    def preprocess(self, docs: List[Document]):
```

Comment:
This is looking good. Some thoughts I have are:
- I'm not sure we need to return `Document` objects here. I think we could also just return a dictionary with the relevant information. Not a huge deal, but I think it is an unnecessary use of a schema object that will only be used internally by this node (the split docs aren't exposed), and already requires meta information to work in the postprocessor. 
- Would it be possible to copy the header info into each sub-table? I think we discussed that this would be desirable. It might not be as relevant for our current project, but for other users who populate the column headers (like giant excel tables) I think this feature would be very useful.
- Is there any way we could automatically determine how many rows (or cols) we should truncate by based on the provided TAPASTokenizer. Otherwise, users will not know what numbers to provide in `num_rows` and `num_cols`.  Additionally, if users provide too large of numbers for `num_rows` and `num_cols` then the sub tables will be truncated by HF so some information will still be lost.

Reply:
> I'm not sure we need to return Document objects here. I think we could also just return a dictionary with the relevant information. Not a huge deal, but I think it is an unnecessary use of a schema object that will only be used internally by this node (the split docs aren't exposed), and already requires meta information to work in the postprocessor.

You mean a dictionary with 'content', 'id', 'row_offset', 'col_offset' instead of a `Document`?

> Would it be possible to copy the header info into each sub-table? I think we discussed that this would be desirable. It might not be as relevant for our current project, but for other users who populate the column headers (like giant excel tables) I think this feature would be very useful.

Currently it preserves the column header as subsetting is based on `iloc`. `num_row_fixed` and `num_col_fixed` will be used as additional first rows/cols to keep fixed/duplicated in all sub-tables.

> Is there any way we could automatically determine how many rows (or cols) we should truncate by based on the provided TAPASTokenizer. Otherwise, users will not know what numbers to provide in num_rows and num_cols. Additionally, if users provide too large of numbers for num_rows and num_cols then the sub tables will be truncated by HF so some information will still be lost.

Yes, it would be possible if we basically tokenize first and then do something similar to `_split_docs_into_passages`. And ya, if not, `num_rows` and `num_cols` will be guesswork for us setting the defaults and the users who want to change it.

Reply:
> You mean a dictionary with 'content', 'id', 'row_offset', 'col_offset' instead of a Document?

Yes just as a consideration. It might be easier to work with and pass around dictionaries. We also aren't technically abiding by the Document schema since I believe each doc id should be unique which isn't what we want in this case. 

> Currently it preserves the column header as subsetting is based on iloc. num_row_fixed and num_col_fixed will be used as additional first rows/cols to keep fixed/duplicated in all sub-tables.

Oh nice! I didn't realize that. Shows my understanding of pandas XD

> Yes, it would be possible if we basically tokenize first and then do something similar to _split_docs_into_passages. And ya, if not, num_rows and num_cols will be guesswork for us setting the defaults and the users who want to change it.

I think this would definitely be the ideal solution, but definitely more work. Maybe worth investigating to see if it is possible to do in this PR, but we could leave this as a future improvement.